packages/gatsby/src/redux/reducers/component-data-dependencies.js

```diff
@@ -1,46 +1,51 @@
-const _ = require(`lodash`)
-
-module.exports = (state = { nodes: {}, connections: {} }, action) => {
+module.exports = (
+  state = { nodes: new Map(), connections: new Map() },
+  action
+) => {
   switch (action.type) {
     case `DELETE_CACHE`:
-      return { nodes: {}, connections: {} }
+      return { nodes: new Map(), connections: new Map() }
     case `CREATE_COMPONENT_DEPENDENCY`:
       if (action.payload.path === ``) {
         return state
       }
 
       // If this nodeId not set yet.
       if (action.payload.nodeId) {
-        let existingPaths = []
-        if (state.nodes[action.payload.nodeId]) {
-          existingPaths = state.nodes[action.payload.nodeId]
+        let existingPaths = new Set()
+        if (state.nodes.has(action.payload.nodeId)) {
+          existingPaths = state.nodes.get(action.payload.nodeId)
+        }
+        if (!existingPaths.has(action.payload.path || action.payload.id)) {
+          existingPaths.add(action.payload.path || action.payload.id)
         }
-        const newPaths = _.uniq(
-          existingPaths.concat(action.payload.path || action.payload.id)
-        )
-        state.nodes[action.payload.nodeId] = newPaths
+        state.nodes.set(action.payload.nodeId, existingPaths)
```

Comment:
I wonder if we could something more here to avoid initialising empty sets, and setting already set fields in set - maybe something like:

```js
let existingPaths = state.nodes.get(action.payload.nodeId)
if (!existingPaths) {
  existingPaths = new Set()
  state.nodes.set(action.payload.nodeId, existingPaths)
}

// not sure if `has` check is even needed here
if (!existingPaths.has(action.payload.path || action.payload.id)) {
  existingPaths.add(action.payload.path || action.payload.id)
}
```

Reply:
and similar for the `connection` case

Reply:
Let's skip this change, it doesn't seem to have any effect and keeping it more 1to1 gives me more confidence